File: banda/IvanRC-banda-Java/src/main/java/factoryMusicosBanda/FactoryMusicos.java
package factoryMusicosBanda;

import modelsBandaMusicos.*;

public class FactoryMusicos {

    public FactoryMusicos(){}
    private static FactoryMusicos instance = null;
    public static FactoryMusicos getInstance(){
        if(instance == null) instance = new FactoryMusicos();
        return instance;
    }

    /**
     * función que sirve para crear un musico con datos aleatorios
     * @return el musico creado con datos aleatorios
     */
    public Musico createMusicoRandom() {
        Musico musico = null;
        int chance = (int)(Math.random()*100);
        String[] nombres = {"Iván","Jia","Javier","Ramón","Belén","Cristobal","Rapado","Elkevin","Mayorca","Jorge","Nefer","Sebastian","Álvaro","Romeo"};
        int añosExperiencia = (int)(Math.random()*38);
        int capacidadPulmonar = (int)(Math.random()*10);
        String[] tonos = {"agudo", "grave", "suave", "profundo"};
        int cantidadTeclas = (int)(Math.random()*24);
        int numeroCuerdas = (int)(Math.random()*8);
        if(chance <= 5){
            musico = new Trompetista(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, capacidadPulmonar);
        }else{
            if(chance <= 25){
                musico = new Cantante(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, tonos[(int)(Math.random()* tonos.length)]);
            }else{
                if(chance <= 45){
                    musico = new Guitarrista(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, tipoGuitarra.values()[(int)(Math.random()* tipoGuitarra.values().length)]);
                }else{
                    if(chance <= 55){
                        musico = new Bajista(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, numeroCuerdas);
                    }else{
                        if(chance <= 65){
                            musico = new Teclista(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, cantidadTeclas);
                        }else{
                            if(chance <= 80){
                                musico = new Percusionista(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, tipoPercusion.values()[(int)(Math.random()* tipoPercusion.values().length)]);
                            }else{
                                if(chance <= 95){
                                    musico = new CantanteGuitarrista(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, tonos[(int)(Math.random()* tonos.length)], tipoGuitarra.values()[(int)(Math.random()* tipoGuitarra.values().length)]);
                                }else{
                                    musico = new Multinstrumental(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, numeroCuerdas, cantidadTeclas, tipoPercusion.values()[(int)(Math.random()* tipoPercusion.values().length)]);
                                }
                            }
                        }
                    }
                }
            }
        }
        return musico;
    }
}

File: banda/IvanRC-banda-Kotlin/src/main/kotlin/factoryMusicosBanda/FactoryMusicos.kt
package factoryMusicosBanda

import modelsBandaMusicos.*

class FactoryMusicos {

    companion object{
        private var instance: FactoryMusicos? = null
        fun getInstance(): FactoryMusicos?{
            if(instance==null) instance= FactoryMusicos()
            return instance
        }
    }

    /**
     * función que sirve para crear un musico con datos aleatorios
     * @return el musico creado con datos aleatorios
     */
    fun createMusicoRandom(): Musico? {
        var musico: Musico? = null
        val chance = (1..100).random()
        val nombres = arrayOf("Iván","Jia","Javier","Ramón","Belén","Cristobal","Rapado","Elkevin","Mayorca","Jorge","Nefer","Sebastian","Álvaro","Romeo")
        val añosExperiencia = (1..38).random()
        val capacidadPulmonar = (1..10).random()
        val tonos = arrayOf("agudo", "grave", "suave", "profundo")
        val cantidadTeclas = (1..24).random()
        val numeroCuerdas = (1..8).random()
        if(chance <= 5){
            musico = Trompetista(nombre = nombres.random(), añosExperiencia = añosExperiencia, capacidadPulmonal = capacidadPulmonar)
        }else{
            if(chance <= 25){
                musico = Cantante(nombre = nombres.random(), añosExperiencia = añosExperiencia, tono = tonos.random())
            }else{
                if(chance <= 45){
                    musico = Guitarrista(nombre = nombres.random(), añosExperiencia = añosExperiencia, tipoDeGuitarra = tipoGuitarra.values().random())
                }else{
                    if(chance <= 55){
                        musico = Bajista(nombre = nombres.random(), añosExperiencia = añosExperiencia, numeroCuerdas = numeroCuerdas)
                    }else{
                        if(chance <= 65){
                            musico = Teclista(nombre = nombres.random(), añosExperiencia = añosExperiencia, cantidadTeclas = cantidadTeclas)
                        }else{
                            if(chance <= 80){
                                musico = Percusionista(nombre = nombres.random(), añosExperiencia = añosExperiencia, tipoDePercusion = tipoPercusion.values().random())
                            }else{
                                if(chance <= 95){
                                    musico = CantanteGuitarrista(nombre = nombres.random(), añosExperiencia = añosExperiencia, tono = tonos.random(), tipoDeGuitarra = tipoGuitarra.values().random())
                                }else{
                                    musico = Multinstrumental(nombre = nombres.random(), añosExperiencia = añosExperiencia, numeroCuerdas = numeroCuerdas, cantidadTeclas = cantidadTeclas, tipoDePercusion = tipoPercusion.values().random())
                                }
                            }
                        }
                    }
                }
            }
        }
        return musico
    }
}